### PR TITLE
Enhancement of documentation for better explaining static/dynamic lin…

### LINKED ISSUE #87: File already exists in database (Windows)
Hi, i'm currently trying to use a packed OSI-Sensor Model as FMU (DummySensor) and execute it via C++ (via FMI4CPP or FMI++) and deserialize (in the same application) under Windows:
It seems that there is an issue, if same protobuf-compiled code (=OSI Library) is shared between applications/dll

- Main Application: links to open_simulation_interface.dll
- Sensor Model FMU: also links to open_simulation_interface.dll

(if i do not include osi headers, it works)
grafik
Also splitting the deserialization part into a seperate DLL does not solve the problem.
seems to be related to: https://stackoverflow.com/questions/37051635/several-shared-object-using-same-proto-leading-the-the-error-file-already-exist
(but did not solve the problem)

Is anyone also experience that kind of problem in C++ Environment?
Shouldn't it be a normal use case in a testing environment e.g. to have a multiple FMU running, e.g. one produces SensorData and one consumes SensorData?

### PR DESCRIPTION
**Pull request by PhRosenberger**
﻿#### Reference to a related issue in the repository
Relates to
- #592
- https://github.com/OpenSimulationInterface/osi-sensor-model-packaging/issues/87
- https://github.com/OpenSimulationInterface/osi-documentation/issues/64

#### Add a description
We need better install instructions for static / dynamic linking, so this PR should enhance the README for this.
More explanation should then follow in the OSI documentation, I guess?

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation) for osi-sensor-model-packaging.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests / Github Actions pass locally with my changes.

### CODE REVIEW

**Comment by jdsika** [comment]
consider placeholder for version

**Comment by PhRosenberger** [comment]
Install instructions are located only at one place in the documentation now. OSMP docu (part) just has to link to it then.

File: doc/setup/build_install_example.adoc [hunk 1]
```diff
@@ -18,8 +19,7 @@ Demonstrates a simple C network proxy that can send and receive OSI data via TCP
 
 **Prerequisites**
 
-* You have installed _cmake_ version 3.10.2 or higher.
-* You have installed _protobuf_ version 3.0.0 or higher.
+* You have installed all prerequisites according to the way of linking of your choice, as described in <<Setting up OSI>>.
```
Comment (pmai) [inline doc/setup/build_install_example.adoc]:
```suggestion
* You have installed all prerequisites according to the way of linking of your choice, as described in xref:setup/setting_up_OSI.adoc#top-setting-up-osi[].
```

As discussed in CCB 2023-03-13, adjustments to OSI repo to make setup section a separate file (@PhRosenberger and @philipwindecker will adjust PR OpenSimulationInterface/open-simulation-interface#715 to create the separate file).

**Review by pmai — CHANGES_REQUESTED** [review]

File: doc/setup/build_install_example.adoc (hunk 1 above)
Comment (philipwindecker) [reply doc/setup/build_install_example.adoc]: My bad: It should be "top" instead of "sec". I will fix this in the suggestion above.

Comment (philipwindecker) [reply doc/setup/build_install_example.adoc]: Actually, I would suggest staying with the original syntax for readability here. This is also in line with the ASAM Editorial Guide.
The full path is helpful when trying to point to a specific file. However, if the anchor is unique anyway, the following is easier to create and use:

```suggestion
* You have installed all prerequisites according to the way of linking of your choice, as described in <<top-setting-up-osi>>.
```

**Review by pmai — APPROVED** [review]

**Comment by pmai** [comment]
CCB on 2023-03-27: Merge as-is.

**Pull request merged**